src/graphql-project/internal/core/domain/employee.go

```diff
@@ -0,0 +1,42 @@
+package domain
+
+import "time"
+
+type EmployeeRole string
+
+const (
+	EMPLOEE EmployeeRole = "EMPLOYEE"
+	MANAGER EmployeeRole = "MANAGER"
+)
+
+type Employee struct {
+	ID         string  `validate:""`
+	Name       string  `validate:"required"`
+	EmpID      string  `validate:"required"`
+	Mobile     *string `validate:""`
+	Email      *string `validate:"required"`
+	Address    *string `validate:""`
+	ManagerID  *string `validate:""`
+	IsActive   *bool   `validate:"required"`
+	Department *string `validate:"required"`
+
+	CreatedAt     time.Time  `validate:"required"`
+	CreatedBy     string     `validate:"required,ascii,max=100"`
+	LastUpdatedAt *time.Time `validate:""`
+	LastUpdatedBy *string    `validate:"omitempty,ascii,max=100"`
+}
+
+func GetEmployee(name, empID, mobile, email, department,
+	add, managerID string, isActive bool) *Employee {
+
+	return &Employee{
+		Name:       name,
+		EmpID:      empID,
+		Mobile:     &mobile,
```

Comment:
These can be passed a pointer to the function?

Reply:
removed not required